### DISCUSSION TITLE
useFieldArray with a required radio group

### DISCUSSION THREAD

**Discussion by ThomasRutzer** [react-hook-form/react-hook-form#10234] [Q&A]
Hi,

I am having a list with the following FormData:
`{ name: string; isSelected: boolean }[]`

Now, I want to make use of `useFieldArray`, but with this requirements:
- Every `isSelected` is rendered as a radio input. The radios should behave like a radio group, i.e. only one can be selected (this is usually accomplished by a similar "name" attribute of all radios)
- set a `required` rule so that exactly one radio must be selected by the user

I have prepared a [Codesandbox](https://codesandbox.io/s/field-array-with-required-radio-group-9w2kc9) and hope somebody can help me achieving this. 

There is also this note in the [official docs](https://react-hook-form.com/api/usefieldarray/), maybe someone has a pointer / example for that:
_Each input name needs to be unique, if you need to build checkbox or radio with the same name then use it with useController or controller._

**Comment by Shaker-Hamdi** [comment]
I think you just need to give the same name for all the `<input type="radio" />` field ...

https://codesandbox.io/s/field-array-with-required-radio-group-9w2kc9?file=/src/App.tsx

**Comment by evgeniy-ferapontov** [comment]
@ThomasRutzer have you ever found a solution?

**Comment by Jamalahmad123** [comment]
Any Updates
